models/src/main/scala/it/pagopa/interop/agreementmanagement/model/agreement/PersistentAgreement.scala

```diff
@@ -12,9 +11,12 @@ final case class PersistentAgreement(
   consumerId: UUID,
   state: PersistentAgreementState,
   verifiedAttributes: Seq[PersistentVerifiedAttribute],
+  certifiedAttributes: Seq[PersistentCertifiedAttribute],
+  declaredAttributes: Seq[PersistentDeclaredAttribute],
```

Comment:
Why not a Set instead of a Seq? So you can remove `distinctBy` from `Adapters`.

Reply:
makes sense for type clarity 👍 

But would you remove the distinct? It is used in the conversion from the request to the persistent model, so if for example `PersistentCertifiedAttribute` will contain a `field2`, and the request contains 2 certified attributes with same `id` but different `field2` value, the `Set` would not remove the duplication. 🤔 

Not that ignoring one of them randomly is an optimal solution..

Reply:
I relaunch: why not `List` instead of `Seq` 😇

Reply:
> But would you remove the distinct? It is used in the conversion from the request to the persistent model, so if for example `PersistentCertifiedAttribute` will contain a `field2`, and the request contains 2 certified attributes with same `id` but different `field2` value, the `Set` would not remove the duplication. 🤔

I got your point.
Ok for the `distinctBy`.

> I relaunch: why not `List` instead of `Seq`? 😇

In these specific cases, I suggest using Set, but I agree with you, that we should use List instead of Seq in the other cases.